socks5.hpp

```diff
@@ -65,7 +65,7 @@ namespace socks5 { // threw in the kitchen sink for error codes
       return instance;
     }
 
-    error_code make_error_code(result_code se) {
+    inline error_code make_error_code(result_code se) {
```

Comment:
Very good points. I think only clang complains about these, and it's a good point.